### ISSUE TITLE
I have a question about the secp256k1 algorithm

### ISSUE THREAD

**Issue by monwazchh** [brichard19/BitCrack#424]
Hi there. i am trying to convert a private key to public key using only GPU power. i have already done this with some code running on CPU. but this is insufficient for me. Because CPU codes have a processing power of around 640 thousand per second. So I need an algorithm that only runs on the GPU.
I'm wondering, in this project, does the public key generation from the private key work on the GPU or on the CPU?
Can anyone help me with this? I don't know much about C++.

**Comment by wisammechano** [comment]
This code repo does what you are referring to both on CPU and GPU. Relevant code for this on GPU can be found in cudaMath part of the project.

**Comment by monwazchh** [comment]
> This code repo does what you are referring to both on CPU and GPU. Relevant code for this on GPU can be found in cudaMath part of the project.

Obviously I had already found the right codes by rummaging through the folders. However, I couldn't figure out how to use them. When I examined the project step by step, I saw that the public key generation from the private key takes place entirely on the CPU. However, there are GPU codes for public key generation in the project files. I am very confused.
The only thing I need is an algorithm that runs only on the GPU and will convert any private key into a public key.

Can you help me with this? 
Normally I know C# but this project is in C++ and it is very confusing for me.

**Comment by nassssro0444** [comment]
> > This code repo does what you are referring to both on CPU and GPU. Relevant code for this on GPU can be found in cudaMath part of the project.
> 
> Obviously I had already found the right codes by rummaging through the folders. However, I couldn't figure out how to use them. When I examined the project step by step, I saw that the public key generation from the private key takes place entirely on the CPU. However, there are GPU codes for public key generation in the project files. I am very confused. The only thing I need is an algorithm that runs only on the GPU and will convert any private key into a public key.
> 
> Can you help me with this? Normally I know C# but this project is in C++ and it is very confusing for me.

i m using an amd gpu and i had the same problem plz if u find any solution notify me

**Comment by monwazchh** [comment]
> > > This code repo does what you are referring to both on CPU and GPU. Relevant code for this on GPU can be found in cudaMath part of the project.
> > 
> > 
> > Obviously I had already found the right codes by rummaging through the folders. However, I couldn't figure out how to use them. When I examined the project step by step, I saw that the public key generation from the private key takes place entirely on the CPU. However, there are GPU codes for public key generation in the project files. I am very confused. The only thing I need is an algorithm that runs only on the GPU and will convert any private key into a public key.
> > Can you help me with this? Normally I know C# but this project is in C++ and it is very confusing for me.
> 
> i m using an amd gpu and i had the same problem plz if u find any solution notify me

Of course, if I find a solution, I will let you know here. Please let me know if you find a solution.

**Comment by ik316m** [comment]
> Hi there. i am trying to convert a private key to public key using only GPU power. i have already done this with some code running on CPU. but this is insufficient for me. Because CPU codes have a processing power of around 640 thousand per second. So I need an algorithm that only runs on the GPU. I'm wondering, in this project, does the public key generation from the private key work on the GPU or on the CPU? Can anyone help me with this? I don't know much about C++.

I am also using C# and NBitcoin but I able to generate only 1000 keys per second, if you can share you C# code please.

**Comment by monwazchh** [comment]
> > Hi there. i am trying to convert a private key to public key using only GPU power. i have already done this with some code running on CPU. but this is insufficient for me. Because CPU codes have a processing power of around 640 thousand per second. So I need an algorithm that only runs on the GPU. I'm wondering, in this project, does the public key generation from the private key work on the GPU or on the CPU? Can anyone help me with this? I don't know much about C++.
> 
> I am also using C# and NBitcoin but I able to generate only 1000 keys per second, if you can share you C# code please.

Nothing special, I just use Parallel Foreach so that all cores of the processor generate addresses in parallel at 100% power. The higher the number of cores and the higher the Ghz, the faster it will generate.
